File: spec-traits/src/main/java/bsp/traits/UntaggedUnionTrait.java
package bsp.traits;

import software.amazon.smithy.model.node.Node;
import software.amazon.smithy.model.shapes.ShapeId;
import software.amazon.smithy.model.traits.AnnotationTrait;

public final class UntaggedUnionTrait extends AnnotationTrait {
	public static ShapeId ID = ShapeId.from("jsonrpc#untaggedUnion");

	public UntaggedUnionTrait() {
		super(ID, Node.objectNode());
	}

	public static final class Provider extends AnnotationTrait.Provider<UntaggedUnionTrait> {
		public Provider() {
			super(ID, (node) -> new UntaggedUnionTrait());
		}
	}
}

File: spec-traits/src/main/kotlin/traits/UntaggedUnionTrait.kt
package traits

import software.amazon.smithy.model.node.ObjectNode
import software.amazon.smithy.model.shapes.ShapeId
import software.amazon.smithy.model.traits.AnnotationTrait

class UntaggedUnionTrait(node: ObjectNode) : AnnotationTrait(ID, node) {
    class Provider : AnnotationTrait.Provider<UntaggedUnionTrait>(ID, ::UntaggedUnionTrait)

    companion object {
        val ID: ShapeId = ShapeId.from("traits#untaggedUnion")
    }
}
